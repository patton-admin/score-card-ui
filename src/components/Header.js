import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { default as React } from "react";
import { NavLink } from "react-router-dom";
import "./../styles/NavBar.css";

const Header = (props) => (
  <div
    className="navbar"
    style={{
      position: "-webkit-sticky",
      position: "sticky",
      top: 0,
      zIndex: 99999,
    }}
  >
    <NavLink
      to="/home"
      exact={true}
      onClick={(e) => e.preventDefault()}
      style={{ pointerEvents: "none" }}
    >
      {" "}
      <FontAwesomeIcon size="1x" icon="lightbulb" color="#15b5ea" /> Patton
      Tracking System{" "}
      <FontAwesomeIcon size="1x" icon="lightbulb" color="#15b5ea" />
    </NavLink>

    <NavLink
      to="/home"
      activeStyle={{
        color: "#15b5ea",
      }}
    >
      <FontAwesomeIcon icon="home" /> Home
    </NavLink>

    <NavLink
      to="/dashboard"
      activeStyle={{
        color: "#15b5ea",
      }}
    >
      <FontAwesomeIcon icon="home" /> Dashboard
    </NavLink>

    <NavLink
      to="/jobOrders"
      activeStyle={{
        color: "#15b5ea",
      }}
    >
      <FontAwesomeIcon icon="shopping-cart" /> Job Orders
    </NavLink>

    <NavLink
      to="/globalBucket"
      activeStyle={{
        color: "#15b5ea",
      }}
    >
      <FontAwesomeIcon icon="globe-americas" /> Score Card
    </NavLink>

    {/* <NavLink
      to="/todo"
      activeStyle={{
        color: "#15b5ea",
      }}
    >
      <FontAwesomeIcon icon="list-ol" /> To-Do's
    </NavLink> */}

    <NavLink
      to="/support"
      activeStyle={{
        color: "#15b5ea",
      }}
    >
      <FontAwesomeIcon icon="info" /> Support
    </NavLink>

    <NavLink
      to="/help"
      activeStyle={{
        color: "#15b5ea",
      }}
    >
      <FontAwesomeIcon icon="hands-helping" /> Help
    </NavLink>

    <NavLink
      to="/logout"
      style={{ float: "right" }}
      onClick={props.handleLogout}
    >
      <FontAwesomeIcon icon="sign-out-alt" />
    </NavLink>

    <NavLink to="/user" style={{ float: "right" }}>
      <FontAwesomeIcon icon="user" />
      {"  "} Hi {props.user !== undefined ? props.user.split("@")[0] : ""}
    </NavLink>
  </div>
);

export default Header;
