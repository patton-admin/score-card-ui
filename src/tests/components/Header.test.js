import { shallow } from "enzyme";
import React from "react";
import Header from "./../../components/Header";

test(`testing the rendering of Header Component`, () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("div").length).toBe(1);
});
