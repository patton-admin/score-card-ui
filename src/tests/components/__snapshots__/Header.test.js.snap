// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of Header Component 1`] = `
<div
  className="navbar"
>
  <NavLink
    exact={true}
    style={
      Object {
        "backgroundColor": "rgba(225, 110, 28, 0.91)",
      }
    }
    to="/"
  >
     
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="lightbulb"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="1x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     Central Hiring System
     
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="lightbulb"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="1x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="/home"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="home"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     Home
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="/support"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="info"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     support
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="/help"
  >
    <FontAwesomeIcon
      activeStyle={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="hands-helping"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    Help
  </NavLink>
  <NavLink
    style={
      Object {
        "float": "right",
      }
    }
    to="/user"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="user"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
      
     Hi 
  </NavLink>
</div>
`;
